packages/usa-in-page-navigation/src/test/in-page-navigation.spec.js

```diff
@@ -48,28 +49,68 @@ tests.forEach(({ name, selector: containerSelector }) => {
 
     let theNav;
     let theList;
+    let originalOffsetTop;
 
     before(() => {
+      originalOffsetTop = Object.getOwnPropertyDescriptor(
+        HTMLElement.prototype,
+        "offsetTop"
+      );
+      Object.defineProperty(HTMLElement.prototype, "offsetTop", {
+        get() {
+          const heading = this.closest("h2,h3");
+
+          let index = 0;
+          let sibling = heading;
+          while (true) {
```

Comment:
This tries to simulate a vertical offset in order to test that `offsetTop` used for scrolling is behaving as expected, since JSDOM doesn't emulate that. The logic essentially considers computes the index of the current heading and multiplies that by a constant number. I'm open to other ideas for how to ensure the scrolling occurs as expected.

Reply:
I see, thanks. I couldn't find any alternative solutions to `offsetTop` in their [issues queue](https://github.com/jsdom/jsdom/issues?q=offsetTop+sort:updated-desc).

I guess what threw me off initially was the `while (true)` inside of `Object.defineProperty`.